### DISCUSSION TITLE
Should the `GET /v0/blob/<id>` API Endpoint Accommodate Range Requests?

### DISCUSSION THREAD

**Discussion by masih** [filecoin-project/motion#14] [General]
It's undeniable that accommodating byte range requests offers significant benefits. However, the feasibility of this feature largely depends on the blob's location and the available methods for its retrieval. 

For instance, facilitating byte range requests over HTTP is straightforward when a blob is cached locally. Yet, challenges arise when data is fetched from SPs that may not support range retrieval.

Considering these factors, is it a necessity for the 'fetch blocks' functionality to facilitate byte range requests, both for the MVP and future iterations?

**Comment by willscott** [comment]
I think we've said 'yes', we should support it at the API, but we can initially be inefficient in fetching the backing data for an MVP iteration and add the complexity of efficiently transferring only requested ranges as a performance feature we can prioritize separately.

**Reply by masih** [reply]
Thanks
